packages/manager/src/components/RegionSelect/RegionMultiSelect.stories.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+
+import { regions } from 'src/__data__/regionsData';
+import { Box } from 'src/components/Box';
+import { Flag } from 'src/components/Flag';
+import {
+  RemovableItem,
+  RemovableSelectionsList,
+} from 'src/components/RemovableSelectionsList/RemovableSelectionsList';
+import { sortByString } from 'src/utilities/sort-by';
+
+import { RegionMultiSelect } from './RegionMultiSelect';
+import { StyledFlagContainer } from './RegionSelect.styles';
+
+import type { RegionMultiSelectProps } from './RegionSelect.types';
+import type { Meta, StoryObj } from '@storybook/react';
+import type { RegionSelectOption } from 'src/components/RegionSelect/RegionSelect.types';
+
+interface SelectedRegionsProps {
+  onRemove: (data: RegionSelectOption) => void;
+  selectedRegions: RegionSelectOption[];
+}
+
+interface LabelComponentProps {
+  selection: RemovableItem;
+}
+
+const sortRegionOptions = (a: RegionSelectOption, b: RegionSelectOption) => {
+  return sortByString(a.label, b.label, 'asc');
+};
+
+const LabelComponent = ({ selection }: LabelComponentProps) => {
+  return (
+    <Box
+      sx={{
+        alignItems: 'center',
+        display: 'flex',
+        flexGrow: 1,
+      }}
+    >
+      <StyledFlagContainer>
+        <Flag country={selection.data.country} />
+      </StyledFlagContainer>
+      {selection.label}
+    </Box>
+  );
+};
+
+const SelectedRegionsList = ({
+  onRemove,
+  selectedRegions,
+}: SelectedRegionsProps) => {
+  const handleRemove = (data: RemovableItem) => {
+    onRemove((data as unknown) as RegionSelectOption);
+  };
```

Comment:
No need to cast here I think

```suggestion
 const handleRemove = (item: RemovableItem) => {
    onRemove(item.data);
 };
```